tests/test_openjdk.py

```diff
@@ -82,6 +84,51 @@ def test_jdk_version(container, java_version):
     )
 
 
+def test_java_home(auto_container: ContainerData):
+    """Check that the environment variable ``JAVA_HOME`` is set to the correct
+    value.
+
+    - check that ``JAVA_HOME`` is a directory
+    - check that ``JAVA_ROOT`` is a directory
+    - check that ``JAVA_BINDIR`` is a directory and that it contains the
+      executable :command:`java`.
+    - parse the output of :command:`java -XshowSettings:properties -version`,
+      extract the setting ``java.home`` and compare it to ``JAVA_HOME``
+    """
+
+    def get_env_var(var: str) -> str:
+        return auto_container.connection.check_output(f"echo ${var}").strip()
+
+    java_home_path = get_env_var("JAVA_HOME")
+    java_home = auto_container.connection.file(java_home_path)
+    assert java_home.exists and java_home.is_directory
+
+    java_root = auto_container.connection.file(get_env_var("JAVA_ROOT"))
+    assert java_root.exists and java_root.is_directory
+
+    java_bindir_path = get_env_var("JAVA_BINDIR")
+    java_bindir = auto_container.connection.file(java_bindir_path)
+    assert java_bindir.exists and java_bindir.is_directory
+
+    java = auto_container.connection.file(
+        os.path.join(java_bindir_path, "java")
+    )
+    assert java.exists and java.is_file and java.mode & stat.S_IXUSR
+
```

Comment:
Since we are here already, I would even execute the binary once, e.g.

```suggestion
auto_container.connection.run_expect([0], java_bindir_path + "/java --version")
```

WDYT?